vm/devices/net/mana_driver/src/gdma_driver.rs

```diff
@@ -161,17 +161,49 @@ const NUM_PAGES: usize = 6;
 // RWQEs have no OOB and one SGL entry so they are always exactly 32 bytes.
 const RWQE_SIZE: u32 = 32;
 
+impl<T: DeviceBacking> GdmaDriver<T> {
+    /// Polls the shared‐memory ownership bit until PF gives it back (or we timeout / device not present).
+    /// Returns `Some(header)` if we successfully see VF ownership (i.e. PF bit cleared),
+    /// or `None` if the device not present or we hit our timeout.
+    fn wait_for_vf_to_own_shmem(&self, log_ctx: &str) -> Option<SmcProtoHdr> {
```

Comment:
instead of passing log context, you can return a `Result<SmcProtoHdr>` and then on the caller end you can do `self.wait_for_vf_to_own_shmem().context(log context here)`